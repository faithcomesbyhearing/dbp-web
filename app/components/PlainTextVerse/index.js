/**
 *
 * PlainTextVerse
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import IconsInText from '../IconsInText';
import customStyle from '../../utils/customVerseStyle';

function PlainTextVerse({
	onMouseUp,
	onMouseDown,
	onHighlightClick,
	onNoteClick,
	verse,
	activeVerse,
	verseIsActive,
	oneVerse,
}) {
	return (
		<span
			onMouseUp={onMouseUp}
			onMouseDown={onMouseDown}
			onClick={onHighlightClick}
			style={customStyle(verse)}
			className={
				verseIsActive &&
				(parseInt(activeVerse, 10) === verse.verse_start ||
					activeVerse === verse.verse_start_alt)
					? 'align-left active-verse'
					: 'align-left'
			}
			data-verseid={verse.verse_start}
			key={verse.verse_start}
		>
			{oneVerse && <br />}
			<sup data-verseid={verse.verse_start}>
				&nbsp;
				{verse.verse_start_alt || verse.verse_start}
				&nbsp;
			</sup>
			<IconsInText
				clickHandler={onNoteClick}
				bookmarkData={{
					hasBookmark: verse.hasBookmark,
					index: verse.bookmarkIndex,
				}}
				noteData={{ hasNote: verse.hasNote, index: verse.noteIndex }}
			/>
			{verse.hasHighlight ? (
				<span
					data-verseid={verse.verse_start}
					dangerouslySetInnerHTML={{ __html: verse.verse_text }}
				/>
			) : (
				<span data-verseid={verse.verse_start}>{verse.verse_text}</span>
			)}
		</span>
	);
}

PlainTextVerse.propTypes = {
	onMouseUp: PropTypes.func,
	onMouseDown: PropTypes.func,
	onHighlightClick: PropTypes.func,
	onNoteClick: PropTypes.func,
	verse: PropTypes.object,
	activeVerse: PropTypes.number,
	verseIsActive: PropTypes.bool,
	oneVerse: PropTypes.bool,
};

export default PlainTextVerse;
