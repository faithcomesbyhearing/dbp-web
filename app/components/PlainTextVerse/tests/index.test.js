import React from 'react';
import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import renderer from 'react-test-renderer';
import PlainTextVerse from '../index';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('../../IconsInText', () => function iconsInTextMock() {
  return <div id="mockIcons">mockIcons</div>;
});

const onMouseUp = jest.fn();
const onMouseDown = jest.fn();
const onHighlightClick = jest.fn();
const onNoteClick = jest.fn();
const verse = {
	book_id: 'GEN',
	book_name: 'Genesis',
	book_name_alt: 'Genesis',
	chapter: 1,
	chapter_alt: '1',
	verse_end: 1,
	verse_end_alt: '1',
	verse_start: 1,
	verse_start_alt: '1',
	verse_text: 'In the beginning God created the heavens and the earth.',
	wholeVerseHighlighted: true,
};
const activeVerse = 0;
const verseIsActive = false;
const oneVerse = false;

describe('PlainTextVerse', () => {
	it('should return correct component', () => {
		const wrapper = Enzyme.mount(
			<PlainTextVerse
				onMouseUp={onMouseUp}
				onMouseDown={onMouseDown}
				onHighlightClick={onHighlightClick}
				onNoteClick={onNoteClick}
				verse={verse}
				activeVerse={activeVerse}
				verseIsActive={verseIsActive}
				oneVerse={oneVerse}
			/>,
		);

		expect(wrapper.find('#mockIcons').length).toEqual(1);
		expect(wrapper.contains(verse.verse_text)).toEqual(true);
	});
	it('Should match the old snapshot', () => {
		const tree = renderer
			.create(
				<PlainTextVerse
					onMouseUp={onMouseUp}
					onMouseDown={onMouseDown}
					onHighlightClick={onHighlightClick}
					onNoteClick={onNoteClick}
					verse={verse}
					activeVerse={activeVerse}
					verseIsActive={verseIsActive}
					oneVerse={oneVerse}
				/>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Should match the old snapshot for oneVerse option', () => {
		const tree = renderer
			.create(
				<PlainTextVerse
					onMouseUp={onMouseUp}
					onMouseDown={onMouseDown}
					onHighlightClick={onHighlightClick}
					onNoteClick={onNoteClick}
					verse={verse}
					activeVerse={activeVerse}
					verseIsActive={verseIsActive}
					oneVerse
				/>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Should render one verse per line if oneVerse is true', () => {
		const wrapper = Enzyme.mount(
			<PlainTextVerse
				onMouseUp={onMouseUp}
				onMouseDown={onMouseDown}
				onHighlightClick={onHighlightClick}
				onNoteClick={onNoteClick}
				verse={verse}
				activeVerse={activeVerse}
				verseIsActive={verseIsActive}
				oneVerse
			/>,
		);
		expect(wrapper.props().oneVerse).toBe(true);
		expect(wrapper.find('#mockIcons').length).toEqual(1);
		expect(wrapper.contains(verse.verse_text)).toEqual(true);
		expect(wrapper.find('br').length).toEqual(1);
	});
});
