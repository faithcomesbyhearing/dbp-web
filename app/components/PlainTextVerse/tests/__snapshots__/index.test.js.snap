// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainTextVerse Should match the old snapshot 1`] = `
<span
  className="align-left"
  data-verseid={1}
  onClick={[MockFunction]}
  onMouseDown={[MockFunction]}
  onMouseUp={[MockFunction]}
  style={
    {
      "background": "linear-gradient(inherit,inherit)",
    }
  }
>
  <sup
    data-verseid={1}
  >
     
    1
     
  </sup>
  <div
    id="mockIcons"
  >
    mockIcons
  </div>
  <span
    data-verseid={1}
  >
    In the beginning God created the heavens and the earth.
  </span>
</span>
`;

exports[`PlainTextVerse Should match the old snapshot for oneVerse option 1`] = `
<span
  className="align-left"
  data-verseid={1}
  onClick={[MockFunction]}
  onMouseDown={[MockFunction]}
  onMouseUp={[MockFunction]}
  style={
    {
      "background": "linear-gradient(inherit,inherit)",
    }
  }
>
  <br />
  <sup
    data-verseid={1}
  >
     
    1
     
  </sup>
  <div
    id="mockIcons"
  >
    mockIcons
  </div>
  <span
    data-verseid={1}
  >
    In the beginning God created the heavens and the earth.
  </span>
</span>
`;
